site/src/pages/TemplatePage/TemplateInsightsPage/TemplateInsightsPage.tsx

```diff
@@ -0,0 +1,360 @@
+                        "& .MuiLinearProgress-bar": {
+                          backgroundColor: usageColors[i],
+                          borderRadius: 999,
+                        },
+                      }}
+                    />
+                    <Box
+                      sx={{
+                        fontSize: 13,
+                        color: (theme) => theme.palette.text.secondary,
+                        width: 200,
+                        flexShrink: 0,
+                      }}
+                    >
+                      {formatTime(usage.seconds)}
+                    </Box>
+                  </Box>
+                )
+              })}
+          </Box>
+        )}
+      </PanelContent>
+    </Panel>
+  )
+}
+
+const Panel = styled(Box)(({ theme }) => ({
+  borderRadius: theme.shape.borderRadius,
+  border: `1px solid ${theme.palette.divider}`,
+  backgroundColor: theme.palette.background.paper,
+  display: "flex",
+  flexDirection: "column",
+}))
+
+type PanelProps = ComponentProps<typeof Panel>
+
+const PanelHeader = styled(Box)(({ theme }) => ({
+  padding: theme.spacing(2.5, 3, 3),
+}))
+
+const PanelTitle = styled(Box)(() => ({
+  fontSize: 14,
+  fontWeight: 500,
+}))
+
+const PanelSubtitle = styled(Box)(({ theme }) => ({
+  fontSize: 13,
+  color: theme.palette.text.secondary,
+}))
+
+const PanelContent = styled(Box)(({ theme }) => ({
+  padding: theme.spacing(0, 3, 3),
+  flex: 1,
+}))
+
+const NoDataAvailable = (props: BoxProps) => {
+  return (
+    <Box
+      {...props}
+      sx={{
+        fontSize: 13,
+        color: (theme) => theme.palette.text.secondary,
+        textAlign: "center",
+        height: "100%",
+        display: "flex",
+        alignItems: "center",
+        justifyContent: "center",
+        ...props.sx,
+      }}
+    >
+      No data available
+    </Box>
+  )
+}
+
+function mapToDAUsResponse(
+  data: TemplateInsightsResponse["interval_reports"],
+): DAUsResponse {
+  return {
+    tz_hour_offset: 0,
+    entries: data.map((d) => {
+      return {
+        amount: d.active_users,
+        date: d.end_time,
+      }
+    }),
+  }
+}
+
+function toTimeFilter(date: Date) {
+  date.setHours(0, 0, 0, 0)
+  const year = date.getUTCFullYear()
+  const month = String(date.getUTCMonth() + 1).padStart(2, "0")
+  const day = String(date.getUTCDate()).padStart(2, "0")
+
+  return `${year}-${month}-${day}T00:00:00Z`
+}
+
+function formatTime(seconds: number): string {
+  if (seconds < 60) {
+    return seconds + " seconds"
+  } else if (seconds >= 60 && seconds < 3600) {
+    const minutes = Math.floor(seconds / 60)
+    return minutes + " minutes"
+  } else {
+    const hours = Math.floor(seconds / 3600)
+    const remainingMinutes = Math.floor((seconds % 3600) / 60)
+    if (remainingMinutes === 0) {
+      return hours + " hours"
+    } else {
+      return hours + " hours, " + remainingMinutes + " minutes"
+    }
+  }
+}
+
+function sevenDaysAgo() {
```

Comment:
nit: weekAgo :)

Reply:
This is a good one. I don't think a "week ago" is a good name for this because you are at Wed and a week ago would be the previous week from Mon to Fri so 7 days ago is different and for this case would be more accurate